core/sql/langman/LmRoutineJavaObj.cpp

```diff
@@ -192,6 +192,7 @@ LmResult LmRoutineJavaObj::invokeRoutineMethod(
           (*emitRowPtr_)(NULL,0,&qs);
         }
     }
+  jni->DeleteLocalRef(jniResult);
```

Comment:
The idea of this code was to have two C++ calls that need to be made directly one after another:

invokeRoutineMethod() // makes JNI call, returns length of output byte strings
getRoutineInvocationInfo() // retrieves the output byte strings

The local refs are created by the first call and used by the second. They are cleaned up either by the next call to invokeRoutineMethod() or by the destructor.

You are right that if a caller does other JNI calls between those two calls they might get stale objects. I can rewrite this and add an extra copy of the objects. Right now this is not a problem, but we can change this if you prefer. Alternatively I can add comments to make it clear these calls have to follow each other, without other JNI calls in-between.